### ISSUE TITLE
TS Enum code generation

### ISSUE THREAD

**Issue by brakim** [RSuter/NSwag#1064]
My current configuration gives me

```
export enum AreaType {
    _0 = 0, 
    _1 = 1, 
    _2 = 2
}
```

Is there some way to change the output to 

```
export enum AreaType {
    All = 0, 
    One = 1, 
    Two = 2
}
```

I tried enabling `DescribeAllEnumsAsStrings()` but that doesn't do the trick either.

**Comment by RSuter** [comment]
If you generate the spec with NSwag, then the property x-enumNames is automatically generated with the names... 
https://github.com/RSuter/NJsonSchema/wiki/Enums

otherwise you need to define the enums with names instead of the values - to do so with NSwag, use the StringEnumConverter or use the global option, see https://github.com/RSuter/NJsonSchema/wiki/JsonSchemaGenerator#supported-c-attributesannotations (Integer vs string enumerations)

However i think you are generating with swashbuckle - so you have to check their documentation...

**Comment by brakim** [comment]
@RSuter you are right, im using Swashbuckle. Any tip where i could find a solution?

**Comment by RSuter** [comment]
1. Go with Swashbuckle: DescribeAllEnumsAsString() seems to be the way to go: https://stackoverflow.com/a/37658715/876814
2. Use NSwag to generate the spec: https://github.com/RSuter/NSwag/wiki/Middlewares and also use DefaultEnumHandling: string 

In both cases you need to register a global StringEnumConverter
